js/app.js

```diff
@@ -59,14 +59,46 @@ var ingredients = {
     MSC: {symbol: "🎣", info: "Marine Stewardship Council"},
 };
 
-var dateFormat = 'YYYY-MM-DD';
+function dateFromString(raw) {
+    if (raw === undefined) {
+        var d = new Date();
+        // remove all time parameters, for easier comparing
+        d.setHours(0, 0, 0, 0);
+        return d;
+    }
+
+    const datePattern = /^(\d{4})-(\d{2})-(\d{2})$/;
+    const [, year, month, day] = datePattern.exec(raw);
+    return new Date(`${month}, ${day} ${year}`);
+}
 
-function getDate() {
-    var date = m.route.param('date');
-    if (date === undefined) {
-        return moment();
+function dateToString(date) {
+    function padNumber(n) {
+        return String(n).padStart(2, "0");
     }
-    return moment(date, dateFormat);
+
+    return `${date.getFullYear()}-${padNumber(date.getMonth() + 1)}-${padNumber(date.getDate())}`;
+}
+
+/**
+ * Source: https://www.w3resource.com/javascript-exercises/javascript-date-exercise-24.php
+ * @param day
+ * @returns {number}
+ */
+function getWeek(dt) {
+    var tdt = new Date(dt.valueOf());
+    var dayn = (dt.getDay() + 6) % 7;
+    tdt.setDate(tdt.getDate() - dayn + 3);
+    var firstThursday = tdt.valueOf();
+    tdt.setMonth(0, 1);
+    if (tdt.getDay() !== 4) {
+        tdt.setMonth(0, 1 + ((4 - tdt.getDay()) + 7) % 7);
+    }
+    return 1 + Math.ceil((firstThursday - tdt) / 604800000);
```

Comment:
The two implementations differ, the one from moment.js for example returns 2, for the week of the 01.01.2021. Which does not make sense, as far as I'm concerned.

While the one I used returns 53 (of year 2020) which should be the correct week we are searching for. 
Since by definition of ISO8601, the first week is the one, containing the first thursday of the year (https://en.wikipedia.org/wiki/ISO_week_date#First_week).

Further, I fixed a problem when the year of the selected date is different from the one used for week calculation. 
I also added a few comments, to make the code more readable.

By the way, how did you measure the performance?

Reply:
I wasn't entirely sure whether they are identical or not. I agree that 2 doesn't make sense as a result for the 01.01.2021 and 53 is more appropriate.

To measure performance I used [https://jsbench.me/](https://jsbench.me/). Performance doesn't really matter here, because this is not called too often. Still, I was curious how big the difference between both versions is.

Reply:
Alright, yeah performance is not critical here.
So i think, this should be ready to merge